src/test/java/mate/academy/dao/impl/UserDaoImplTest.java

```diff
@@ -0,0 +1,64 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoImplTest extends AbstractTest {
+    private static final String VALID_EMAIL = "bob@i.ua";
+    private static final String INVALID_EMAIL = "alice@i.ua";
+    private static final String VALID_PASSWORD = "12345678";
+    private User user;
+    private Role role;

```

Comment:
you're using this variable only in one method, make it local. Look through other classes and make fields as local variables if it's possible